Jenkinsfile.integration

```diff
@@ -56,12 +57,17 @@ pipeline {
                     sh "git fetch --no-tags"
                     List<String> sourceChanged = sh(returnStdout: true, script: "git diff --name-only origin/master..origin/${env.BRANCH_NAME}").split()
                     echo "Files changed for this PR:\n${sourceChanged.join('\n')}"
+                    FullDeploymentNeeded = false
```

Comment:
this will make it not run when playbooks+docs have changed or tox+docs have changed

Reply:
Why not. It will be set to `true` on line 70 in that case.

Reply:
but if the docs AND the playbooks have changed we want a full CI job no?

Reply:
> Why not. It will be set to `true` on line 70 in that case.

it wont. If the docs and playbooks have changed, it wont enter the `else` and it will skip a full deployment.

We want `FullDeploymentNeeded` when only docs have changed, any other combination should result in running the full job.

Reply:
> it wont. If the docs and playbooks have changed, it wont enter the else and it will skip a full deployment.

It's inside the loop  interating of the list of changed files. So a if there is a single file that is not inside the `doc/` directory, it will get to that else branch. Even if it also matches one of the other conditions.

> We want FullDeploymentNeeded when only docs have changed, any other combination should result in running the full job.

We want `FullDeplyomentNeeded = false` when all changes are limited to the doc dir. We want `FullDeploymentNeeded = true`, when either we're builidng from the master branch (that's why the value defaults to `true` on line#7 and is set to `false` before the loop on line#60) or anything outside the `doc/` dir was changed (that's what the `else` on line#70 should catch).

If you have any better suggestion how to express that in groovy I am all ears :).

Reply:
You are correct!